fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/FiatAccessDeniedExceptionHandler.java

```diff
@@ -54,29 +62,47 @@ public void handleAccessDeniedException(
 
     String errorMessage =
         FiatPermissionEvaluator.getAuthorizationFailure()
-            .map(this::authorizationFailureMessage)
+            .map(authorizationFailure -> authorizationFailureMessage(authorizationFailure, e))
             .orElse("Access is denied");
 
     response.sendError(HttpStatus.FORBIDDEN.value(), errorMessage);
   }
 
   private String authorizationFailureMessage(
-      FiatPermissionEvaluator.AuthorizationFailure authorizationFailure) {
+      FiatPermissionEvaluator.AuthorizationFailure authorizationFailure, AccessDeniedException e) {
+    StringJoiner sj = new StringJoiner(" ");
+
+    defaultErrorDecoration(sj, authorizationFailure);
+
+    // TODO(jonsie): Once we have migrated the current fiat-api code to kork-authz we can produce a
+    //  proper fiat-api module that will allow us to export a class like the one below (instead
+    //  of the current hack which is to pull this in from kork-api).
```

Comment:
Curiosity: What's stopping us?

Reply:
`fiat-api` currently pulls in code from `fiat-core`.  I have to untangle that web, which is not trivial.  That's basically the first step.